src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/amd64/AMD64.java

```diff
@@ -85,9 +85,7 @@ public class AMD64 extends Architecture {
 
     public static final Register[] cpuRegisters = {
         rax, rcx, rdx, rbx, rsp, rbp, rsi, rdi,
-        r8, r9, r10, r11, r12, r13, r14, r15,
-        r16, r17, r18, r19, r20, r21, r22, r23,
-        r24, r25, r26, r27, r28, r29, r30, r31
+        r8, r9, r10, r11, r12, r13, r14, r15
```

Comment:
Hi @dougxc , A comment over cpuRegisters definition relating to its usage will be helpful.

Reply:
I added a brief comment. I will leave it for [JDK-8334717](https://bugs.openjdk.org/browse/JDK-8334717) to expand upon this.